### ISSUE TITLE
project version conflict issue in setup.py 

### ISSUE THREAD

**Issue by ankush109** [collective/cookiecutter-plone-starter#114]
I hope this message finds you well. I recently encountered an issue while setting up the plone/volto project locally using the plone-starter. Upon running the installation with make install, I encountered a version error, which I have outlined below:

![image](https://github.com/collective/cookiecutter-plone-starter/assets/72347734/6660b86a-6814-4a68-b032-50caa00a27c5)

debugging the issue I found out that in the backend/src/setup.py the version was 1.0.0a1 but in the constraints which was given it required the version 4.2.0 hence the previous error was shown , changing this version to the required version 4.2.0 worked and it installed all the packages for the plone project and I was able to run it locally without any errors

![image](https://github.com/collective/cookiecutter-plone-starter/assets/72347734/aec42458-14bd-4f2b-ba35-604a4aa044f4)

**Comment by davisagli** [comment]
This happened because you had the bad luck to pick a package name that collided with one of the core Plone packages (plone.volto). I recommend picking a name that reflects the project you are working on using volto, instead of naming your own project after the tool.

Perhaps the cookiecutter should validate to make sure that the package name isn't the same as one of the core Plone packages.

In the future, please copy and paste text rather than sharing images that include text. They make it hard to read, search, and select text for copying.

**Comment by ankush109** [comment]
ohh I see but It should be mentioned in the installation guide  as this might be a problem for anyone trying to setup locally 
anyways thanks!

**Comment by stevepiercy** [comment]
> Perhaps the cookiecutter should validate to make sure that the package name isn't the same as one of the core Plone packages.

This is the way.

Meanwhile where is a list of core Plone package names maintained? We could add an important admonition in the installation docs to look at that URL.

**Comment by stevepiercy** [comment]
@ankush109 I copied [what you posted in the Community Forum](https://community.plone.org/t/error-in-installing-plone-frontend-and-backend/18482), cleaned up:

```console
ERROR: Cannot install plone-volto 1.0.0a1 (from /home/ankush/Documents/gsoc_2024/plone-volto/backend/src/plone_volto) because these package versions have conflicting dependencies.

The conflict is caused by:
    The user requested plone-volto 1.0.0a1 (from /home/ankush/Documents/gsoc_2024/plone-volto/backend/src/plone_volto)
    The user requested (constraint) plone-volto==4.2.0

To fix this you could try to:
1. loosen the range of package versions you've specified
2. remove package versions to allow pip attempt to solve the dependency conflict

ERROR: ResolutionImpossible: for help visit 
make[2]: *** [Makefile:112: build-dev] Error 1
make[2]: Leaving directory '/home/ankush/Documents/gsoc_2024/plone-volto/backend'
make[1]: *** [Makefile:46: install-backend] Error 2
make[1]: Leaving directory '/home/ankush/Documents/gsoc_2024/plone-volto'
```

See also https://meta.stackoverflow.com/questions/285551/why-should-i-not-upload-images-of-code-data-errors

**Comment by ankush109** [comment]
>> Meanwhile where is a list of core Plone package names maintained? We could add an important admonition in the installation docs to look at that URL.

if I didnt get you wrong may be this is the one https://dist.plone.org/release/6.0.9/constraints.txt

**Comment by davisagli** [comment]
Yeah, the constraints file is more than just Plone packages (also it includes Plone's dependencies) but that's fine for this purpose; we don't want someone using the cookiecutter to use any of those names.

**Comment by ankush109** [comment]
> @ankush109 I copied [what you posted in the Community Forum](https://community.plone.org/t/error-in-installing-plone-frontend-and-backend/18482), cleaned up:
> 
> ```
> ERROR: Cannot install plone-volto 1.0.0a1 (from /home/ankush/Documents/gsoc_2024/plone-volto/backend/src/plone_volto) because these package versions have conflicting dependencies.
> 
> The conflict is caused by:
>     The user requested plone-volto 1.0.0a1 (from /home/ankush/Documents/gsoc_2024/plone-volto/backend/src/plone_volto)
>     The user requested (constraint) plone-volto==4.2.0
> 
> To fix this you could try to:
> 1. loosen the range of package versions you've specified
> 2. remove package versions to allow pip attempt to solve the dependency conflict
> 
> ERROR: ResolutionImpossible: for help visit 
> make[2]: *** [Makefile:112: build-dev] Error 1
> make[2]: Leaving directory '/home/ankush/Documents/gsoc_2024/plone-volto/backend'
> make[1]: *** [Makefile:46: install-backend] Error 2
> make[1]: Leaving directory '/home/ankush/Documents/gsoc_2024/plone-volto'
> ```
> 
> See also https://meta.stackoverflow.com/questions/285551/why-should-i-not-upload-images-of-code-data-errors

yes I will make sure of that in the future sorry !

**Comment by davisagli** [comment]
One note -- pip does some normalization so `plone.volto` and `plone-volto` are considered the same package.

**Comment by stevepiercy** [comment]
Docs PR here: https://github.com/plone/documentation/pull/1597

I would leave this issue open as a reminder to try to do error checking in the cookiecutter and remove the changes in documentation when it is done.
